.github/workflows/ci-pipeline.yml

```diff
@@ -46,3 +46,52 @@ jobs:
 
       - name: Deploy to Azure Web App
         run: curl -dH -X POST ${{ secrets.AZURE_WEB_HOOK }}
+
+  azure-deploy:
+    name: deploy to Azure Web App
+    runs-on: ubuntu-latest
+    needs: docker-publish
+    steps:
+      - name: Deploy to Azure Web App
+        run: curl -dH -X POST ${{ secrets.AZURE_WEB_HOOK }}
+
+
+
+  terraform-deploy:
+    name: deploy using terraform
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged
```

Comment:
I think this job would make most sense to occur after the `docker-publish` step, since we want the latest Docker image to be available before we try to trigger the webhook (otherwise we won't actually release the latest version)